src/zero/module/aou.clj

```diff
@@ -1,23 +1,400 @@
+                               :sample_alias "NA12878",
+                               :green_idat_cloud_path "gs://broad-gotc-test-storage/arrays/HumanExome-12v1-1_A/idats/7991775143_R01C01/7991775143_R01C01_Grn.idat",
+                               :params_file "gs://broad-gotc-test-storage/arrays/HumanExome-12v1-1_A/inputs/7991775143_R01C01/params.txt",
+                               :gender_cluster_file "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/HumanExomev1_1_gender.egt"}))
+  (really-submit-one-workflow :aou-dev {:chip_well_barcode "7991775143_R01C01",
+                                        :bead_pool_manifest_file "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/HumanExome-12v1-1_A.bpm",
+                                        :analysis_version_number 1,
+                                        :extended_chip_manifest_file "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/HumanExome-12v1-1_A.1.3.extended.csv",
+                                        :red_idat_cloud_path "gs://broad-gotc-test-storage/arrays/HumanExome-12v1-1_A/idats/7991775143_R01C01/7991775143_R01C01_Red.idat",
+                                        :zcall_thresholds_file "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/IBDPRISM_EX.egt.thresholds.txt",
+                                        :reported_gender "Female",
+                                        :cluster_file "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/HumanExomev1_1_CEPH_A.egt",
+                                        :sample_lsid "broadinstitute.org:bsp.dev.sample:NOTREAL.NA12878",
+                                        :sample_alias "NA12878",
+                                        :green_idat_cloud_path "gs://broad-gotc-test-storage/arrays/HumanExome-12v1-1_A/idats/7991775143_R01C01/7991775143_R01C01_Grn.idat",
+                                        :params_file "gs://broad-gotc-test-storage/arrays/HumanExome-12v1-1_A/inputs/7991775143_R01C01/params.txt",
+                                        :gender_cluster_file "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/HumanExomev1_1_gender.egt"}))
+
+#_(defn update-workload!
+    "Use transaction TX to update WORKLOAD statuses."
+    [tx workload])
 
 (defn add-workload!
-  "Add the workload described by BODY to the database DB."
+  "Use transaction TX to add the workload described by BODY to the database DB.
+   Due to the continuous nature of the AoU dataflow, this function will only
+   create a new workload table if it does not exist otherwise append records
+   to the existing one."
   [tx body]
-  (->> body
-       first
-       (filter second)
-       (into {})))
+  (let [{:keys [creator cromwell pipeline project]} body
+        {:keys [release top]} workflow-wdl
+        {:keys [commit version]} (zero/get-the-version)
+        probe-result (jdbc/query tx ["SELECT * FROM workload WHERE project = ? AND wdl = ? AND release = ?"
```

Comment:
Why query by the `wdl` field instead of the `pipeline`? The `workflow-wdl` :top has changed independently of the :version before when we refactored the WFL build process so I'm not sure how much we want to rely on that 🤔

Reply:
you are right, I think query `pipeline` is much safer. changed.